src/app/pages/Collectibles/use-collectible-selling.hook.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useState } from 'react';
+
+import BigNumber from 'bignumber.js';
+
+import type { CollectibleDetails } from 'app/store/collectibles/state';
+import { getObjktMarketplaceContract } from 'lib/apis/objkt';
+import { fromFa2TokenSlug } from 'lib/assets/utils';
+import { useAccount, useTezos } from 'lib/temple/front';
+import { getTransferPermissions } from 'lib/utils/get-transfer-permissions';
+import { parseTransferParamsToParamsWithKind } from 'lib/utils/parse-transfer-params';
+
+const DEFAULT_OBJKT_STORAGE_LIMIT = 350;
+
+export const useCollectibleSelling = (assetSlug: string, details?: CollectibleDetails) => {
+  const tezos = useTezos();
+  const { publicKeyHash } = useAccount();
+  const [isSelling, setIsSelling] = useState(false);
+
+  const initiateSelling = useCallback(async () => {
+    const offer = details?.highestOffer;
+    if (!offer || isSelling) return;
+    setIsSelling(true);
+
+    const { contract: tokenAddress, id } = fromFa2TokenSlug(assetSlug);
+    const tokenId = Number(id.toString());
+
+    const contract = await getObjktMarketplaceContract(tezos, offer.marketplace_contract);
+
+    const transferParams = (() => {
+      if ('fulfill_offer' in contract.methods) {
+        return contract.methods.fulfill_offer(offer.bigmap_key, tokenId).toTransferParams();
+      } else {
+        return contract.methods.offer_accept(offer.bigmap_key).toTransferParams();
+      }
+    })();
```

Comment:
```suggestion
    const transferMethod = 'fulfill_offer' in contract.methods
      ? contract.methods.fulfill_offer(offer.bigmap_key, tokenId)
      : contract.methods.offer_accept(offer.bigmap_key);
    const transferParams = transferMethod.toTransferParams();
```
Please check the suggestion with prettier before applying